src/csvToJson.js

```diff
@@ -41,21 +38,22 @@ const generateJsonFiles = (i, fileName, start, stop) => {
 // @TODO update this method later, when we'll migrate to `write` from generator
 // @TODO as this method using "generateJsonFiles" method - it should be updated.
 // or maybe move it into generator file, etc.
-const splitJsonIntoFiles = (fileName) => {
+const splitJsonIntoFiles = (fileName, maxEntriesPerFile, numberOfFiles, dataEntries) => {
   // @TODO add if env.development and use console.log(xxx)
   console.log('---splitJson started---');
-  for (let i; i <= numberOfFiles; i += 1) {
-    const start = (i - 1) * maxEntries;
-    let stop = i * maxEntries;
+  for (let i = 0; i < numberOfFiles; i++) { 
+    const start = i * maxEntriesPerFile; 
+    let stop = ((i+1) * maxEntriesPerFile) - 1;
+    //0-9999,10000-19999,20000...etc
 
-    if (i === numberOfFiles) {
-      stop = result.length + 1;
-      generateJsonFiles(i, fileName, start, stop);
-      return;
+    if (i+1 == numberOfFiles) { //last file
+      stop = result.length - 1;
+      generateJsonFiles(i, fileName, start, stop, dataEntries);
+      return; //end the for loop here
     }
-
-    // @TODO is this related to else statement, confusing ...
-    generateJsonFiles(i, fileName, start, stop);
+.
+    //seems like this is the statement that is usually run in this for loop
```

Comment:
//@TODO just add else statement and nobody will be confused. Short cuts in JS sucks. I mean they cool, but ...